File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        int cnt;
        boolean flag = false;
        for(int i = 0; i <s.length(); i++){
            int chr = s.charAt(i) - 97;
            for(int j = 0; j < index; j++){
                if(!flag){
                    chr = (chr + 1) % 26;    
                }
                if(skip.contains(""+(char)(chr + 97))){
                    chr = (chr + 1) % 26;
                    System.out.println("드감");
                    j--;
                    flag = true;
                }else{
                    flag = false;
                }
            }
            answer = answer + (char)(chr+ 97);
        }
        /*
        index길이만큼 반복문
            a-97/26 이 skip에 속해있나
                있다면 a+1 하고 i--
                없다면
        answer에 더하기
        */
        
        return answer;
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    let answer = '';
    let a = 'z';
    // if(a === 'z'){
    //     console.log('a');
    // }
    
    // console.log(String.fromCharCode(b.charCodeAt(0)));
    for(x of s){
        let al = x;
        for(let i = 0; i<index; i++){
            if(al === 'z'){
                al = 'a';
            }else{
                al = String.fromCharCode(al.charCodeAt(0)+1);
            }
            if(skip.includes(al)){
                i--;
            }
        }
        answer = answer + al;
    }
    return answer;
}